### PR TITLE
164796849 Password Reset Functionality

### PR DESCRIPTION
**Pull request by Simplemart17**
#### Description
Allows users to request for password reset
#### Type of change
- [ ] Bug Fix (a non-breaking change which fixes an issue)
- [x] New Feature (a non-breaking change that adds functionality)
- [ ] New Chore (a non-breaking change that adds functionality)
- [ ] Breaking Change (a fix or feature which would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
#### How Has This Been Tested?
- [ ] End to End
- [x] Integration
#### Checklist:
- [x] Unit test
- [ ] Code coverage
#### Thought Process
- Add form component
- Connect to redux state
- Test page component
- Test actions, reducer
#### Screen Shot
Screenshot 2019-05-14 at 6 27 51 PM
Screenshot 2019-05-18 at 8 47 06 PM
Screenshot 2019-05-18 at 8 47 39 PM
Screenshot 2019-05-18 at 8 48 43 PM

#### PT
[#164796849](https://www.pivotaltracker.com/story/show/164796849)

### CODE REVIEW

File: src/lib/forgotPassword.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import axios from './axios';
+
+export const userForgotPassword = async email => {
+  const forgotPassword = await axios.post('/forgot-password', { email });
+  return forgotPassword;
+};
```
Comment (micah-akpan) [inline src/lib/forgotPassword.js]: Can you use a try/catch here, since you are doing some asynchronous operations here?

**Review by micah-akpan — COMMENTED** [review]
@Simplemart17 Nice work. Can you make the suggested changes, so I can approve it?

File: src/pages/ResetPassword.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    show: true
+  });
+
+  const [error, setError] = useState([]);
+
+  const validate = data => {
+    const errors = {};
+    if (!/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\\$%\\^&\\*])(?=.{8,})/.test(data.password)) {
+      errors.password = 'Password should be At least a lowercase character, uppercase character, numeric character, special character and eight characters long.';
+    }
```
Comment (ssanusi) [inline src/pages/ResetPassword.jsx:26]: @Simplemart17 nice work I think you might want to put all your validation in util folder

Comment (Simplemart17) [reply src/pages/ResetPassword.jsx:26]: Thank you for the feedback, I don't think it might be necessary considering the fact that it is a line of code.

**Review by chikeozulumba — DISMISSED** [review]
@micah-akpan @Simplemart17 LGTM

**Review by Temmyogunbo — DISMISSED** [review]

**Pull request merged**